js/main.js

```diff
@@ -9,24 +9,49 @@ function handleInput(event) {
   previewPhoto.setAttribute('src', urlObj.$url);
 }
 
-const sub = document.querySelector('.submit');
-sub.addEventListener('click', function () {
+form.addEventListener('submit', function (event) {
+  event.preventDefault();
   const submitObj = {
     $title: form.elements.title.value,
     $url: form.elements.url.value,
     $notes: form.elements.notes.value
   };
-  submitObj.nextId = data.nextEntryId++;
-  data.entries.unshift(submitObj);
-  ul.prepend(renderEntry(submitObj));
-  viewSwap('entries');
-  toggleNoEntries();
-  previewPhoto.setAttribute('src', '/images/placeholder-image-square.jpg');
+  if (data.editing === null) {
+    submitObj.entryId = data.nextEntryId++;
+    data.entries.unshift(submitObj);
+    ul.prepend(renderEntry(submitObj));
+    viewSwap('entries');
+    toggleNoEntries();
+  }
+  if (data.editing !== null) {
+    const li = document.querySelectorAll('li');
+    const updatedForm = {
+      $title: form.elements.title.value,
+      $url: form.elements.url.value,
+      $notes: form.elements.notes.value
+    };
+    updatedForm.entryId = data.editing.entryId;
+    for (let i = 0; i < data.entries.length; i++) {
+      if (data.editing.entryId === data.entries[i].entryId) {
+        data.entries[i] = updatedForm;
+        li[i].replaceWith(renderEntry(updatedForm));
+        data.editing = null;
```

Comment:
add a `break` statement in this for loop so that it will stop looping once it finds a match.